packages/next/src/client/components/app-router.tsx

```diff
@@ -362,14 +362,24 @@ function Router({
       forward: () => window.history.forward(),
       prefetch: (href, options) => {
         // Don't prefetch for bots as they don't navigate.
+        if (isBot(window.navigator.userAgent)) {
+          return
+        }
+
+        let url: URL
+        try {
+          url = new URL(addBasePath(href), window.location.href)
+        } catch (_) {
+          throw new Error(
```

Comment:
it should only throw in dev mode

Reply:
This part would have previously also thrown in prod, that's the part i was talking about preserving -- don't want to change this behavior in a patch. This part only rethrows an error with a nicer error message. The fix for the Link crash is that Link wasn't catching this correctly